repose-aggregator/components/filters/header-normalization-filter/src/main/scala/org/openrepose/filters/headernormalization/HeaderNormalizationFilter.scala

```diff
@@ -96,18 +118,56 @@ class HeaderNormalizationFilter @Inject()(configurationService: ConfigurationSer
   }
 
   override def configurationUpdated(config: HeaderNormalizationConfig): Unit = {
-    this.config = config.getHeaderFilters.getTarget.asScala map { target =>
-      val access = if (target.getBlacklist.isEmpty) WhiteList else BlackList
-      val headers = (access match {
-        case WhiteList => target.getWhitelist
-        case BlackList => target.getBlacklist
-      }).get(0).getHeader.asScala.map(_.getId).toSet
-
-      Target(
-        Option(target.getUriRegex).getOrElse(".*").r.pattern,       // if not configured, default is ".*"
-        target.getHttpMethods.asScala.map(_.toString).padTo(1, AllHttpMethods).toSet,  // default is "ALL"
-        access,
-        headers)
+    def getTarget(target: ConfigTarget, accessList: AccessList, headers: HttpHeaderList): Option[Target] = {
+      Option(Target(
+        Option(target.getUriRegex).getOrElse(".*").r.pattern, // if not configured, default is ".*"
+        target.getHttpMethods.asScala.map(_.toString).padTo(1, AllHttpMethods).toSet, // default is "ALL"
+        accessList,
+        headers.getHeader.asScala.map(_.getId).toSet))
```

Comment:
I did put the `.toLowerCase` in here so that it was done at update time instead of every check.